azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -414,7 +451,113 @@ private void setupKeyStore() throws ExecutorManagerException {
         throw new ExecutorManagerException("Failed to delete the cert file");
       }
     }
+  }
+
+  public void cancelFlow(final int execId, final String user)
```

Comment:
Similar to the previous comment about FlowContainerServlet. There is overlap in implementation of cancelFlow, readFlowLog, readJobLog, readJobMetadata with the ones in AzkabanExecutorServer and FlowRunnerManager. 
Please see the class can be structured to minimize the overlap.